Target ci-all fails for fiat_crypto_legacy

### Description of the problem

`make ci-all` fails for target ci-fiat_crypto_legacy
```
File "./src/Util/NumTheoryUtil.v", line 13, characters 0-173:
Error: Required library Coqprime.PrimalityTest.Euler matches several files in
path (found
/home/user/rocq/_build_ci/fiat_crypto_legacy/coqprime/src/Coqprime/PrimalityTest/Euler.vo and
/home/user/rocq/_build/install/default/lib/coq/user-contrib/Coqprime/PrimalityTest/Euler.vo).

Command exited with non-zero status 1
src/Util/NumTheoryUtil.vo (real: 0.24, user: 0.17, sys: 0.06, mem: 342608 ko)
make[1]: *** [Makefile.coq:814: src/Util/NumTheoryUtil.vo] Error 1
make[1]: *** [src/Util/NumTheoryUtil.vo] Deleting file 'src/Util/NumTheoryUtil.glob'
make[1]: Leaving directory '/home/user/rocq/_build_ci/fiat_crypto_legacy'
```
I had to fix fiat_crypto to use latest commit of kami to get to that point, is ci-all even supposed to be run?
Is there an other way to easily run all ci targets?

### Small Rocq / Coq file to reproduce the bug

```coq

```

### Version of Rocq / Coq where this bug occurs

9.0.0

### Interface of Rocq / Coq where this bug occurs

_No response_

### Last version of Rocq / Coq where the bug did not occur

_No response_

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -777,6 +777,8 @@ library:ci-fiat_crypto_legacy:
   needs:
   - build:edge+flambda
   - library:ci-stdlib+flambda
+  - library:ci-coqprime
+  stage: build-3+
   timeout: 1h 30min
 
 # We cannot use flambda due to

--- a/Makefile.ci
+++ b/Makefile.ci
@@ -167,7 +167,7 @@ ci-bedrock2: ci-coqutil ci-riscv_coq ci-kami
 ci-bedrock2_examples: ci-bedrock2
 ci-rupicola: ci-bedrock2 ci-coqutil
 ci-fiat_crypto: ci-coqprime ci-rewriter ci-bedrock2 ci-coqutil ci-rupicola
-ci-fiat_crypto_legacy: ci-stdlib
+ci-fiat_crypto_legacy: ci-coqprime
 ci-fiat_crypto_ocaml: ci-fiat_crypto
 ci-fiat_parsers: ci-stdlib
 

--- a/dev/ci/scripts/ci-fiat_crypto_legacy.sh
+++ b/dev/ci/scripts/ci-fiat_crypto_legacy.sh
@@ -24,8 +24,11 @@ targets2=(
     selected-specific-display
 )
 
+# Use external dependencies to avoid building coqprime from scratch
+make_args=(EXTERNAL_DEPENDENCIES=1)
+
 export COQEXTRAFLAGS='-native-compiler no'
 ( cd "${CI_BUILD_DIR}/fiat_crypto_legacy"
-  make "${targets1[@]}"
-  make -j 1 "${targets2[@]}"
+  make "${make_args[@]}" "${targets1[@]}"
+  make -j 1 "${make_args[@]}" "${targets2[@]}"
 )

